File: Top-Interview-150/Two-Pointers/Two-Sum-II.py
# Problem: Two Sum II
# Technique: Two Pointers
# Link: https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/?envType=study-plan-v2&envId=top-interview-150

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        start = 0
        end = len(numbers) - 1

        while start < end:
            curr_sum = numbers[start] + numbers[end]
            if curr_sum == target:
                return [start + 1,  end + 1]
            elif curr_sum < target:
                start += 1
            else:
                end -= 1

File: Top-Interview-150/Two-Pointers/Two-Sum-II.rs
// Problem: Two Sum II
//Technique: Two Pointers
// Link: https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/?envType=study-plan-v2&envId=top-interview-150

impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let mut left = 0;
        let mut right = numbers.len() - 1;

        while left < right {
            match numbers[left] + numbers[right] {
                sum if sum == target => return vec![(left + 1) as i32, (right + 1) as i32],
                sum if sum < target => left += 1,
                _ => right -= 1,
            }
        }

        Vec::new()
    }
}
